modules/tinymce/src/themes/silver/test/ts/browser/window/SilverDialogSliderApiTest.ts

```diff
@@ -0,0 +1,58 @@
+import { FocusTools, Keys, TestStore, Waiter } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { SugarDocument } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import { Dialog } from 'tinymce/core/api/ui/Ui';
+
+import * as DialogUtils from '../../module/DialogUtils';
+
+describe('browser.tinymce.themes.silver.window.SilverDialogSliderApiTest', () => {
+  const store = TestStore();
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const dialogSpec: Dialog.DialogSpec<{ sliderField: number }> = {
+    title: 'Silver Dialog Test Slider Api',
+    body: {
+      type: 'panel',
+      items: [
+        {
+          type: 'slider',
+          name: 'sliderField',
+          label: 'Slider label',
+          min: 0,
+          max: 100
+        }
+      ]
+    },
+    initialData: {
+      sliderField: 50
+    },
+    onChange: () => {
+      store.add('logOnChangeEvent');
+    }
+  };
+
+  Arr.each([
+    { label: 'Modal', params: { }},
+    { label: 'Inline', params: { inline: 'toolbar' as 'toolbar' }}
+  ], (test) => {
+    context(test.label, () => {
+      it('TINY-10428: dialog onChange handler should fire when using arrow key on slider', async () => {
+        store.clear();
+        const editor = hook.editor();
+        DialogUtils.open(editor, dialogSpec, test.params);
+        await TinyUiActions.pWaitForDialog(editor);
+
+        await FocusTools.pTryOnSelector('Focus should be on slider rail', SugarDocument.getDocument(), '.tox-slider__rail');
+        TinyUiActions.keydown(editor, Keys.left());
+
+        await Waiter.pTryUntil('', () => store.assertEq('Check if on change event is logged after using arrow key', [ 'logOnChangeEvent' ]));
```

Comment:
Not sure this needs to be a waiter and if it needs to be one then add a label for when then waiter fails like `Waited for change event` or something.

Reply:
It passed locally without the waiter, so I am removing it. Thanks!